File: 398. Random Pick Index/C++/Solution.cpp
class Solution {
public:
    vector<int> nums;

    Solution(vector<int> nums) {
        this->nums = nums;
    }

    int pick(int target) {
        int count = 1;
        int result = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == target) {
                if (rand() % count == 0) {
                    result = i;
                }
                count++;
            }
        }
        return result;
    }
};

/**
 * Your Solution object will be instantiated and called as such:
 * Solution obj = new Solution(nums);
 * int param_1 = obj.pick(target);
 */

File: 398. Random Pick Index/Python/Solution.py
class Solution(object):

    def __init__(self, nums):
        """

        :type nums: List[int]
        :type numsSize: int
        """
        self.nums = nums

    def pick(self, target):
        """
        :type target: int
        :rtype: int
        """
        count = 1
        result = None
        for index, num in enumerate(self.nums):
            if num == target:
                if random.randrange(count) == 0:
                    result = index
                count += 1
        return result


# Your Solution object will be instantiated and called as such:
# obj = Solution(nums)
# param_1 = obj.pick(target)
